packages/office-ui-fabric-react/src/components/Button/ActionButton/ActionButton.styles.ts

```diff
@@ -1,79 +1,74 @@
-
-export const getStyles = memoizeFunction((
-  theme: ITheme,
-  customStyles?: IButtonStyles
-): IButtonStyles => {
-  let baseButtonStyles: IButtonStyles = getBaseButtonStyles(theme);
-  let actionButtonStyles: IButtonStyles = {
-    root: {
-      padding: DEFAULT_PADDING,
-      height: DEFAULT_BUTTON_HEIGHT,
-      color: theme.palette.neutralPrimary,
-      backgroundColor: 'transparent'
-    },
-
-    rootHovered: {
-      color: theme.palette.themePrimary,
-    },
-
-    iconHovered: {
-      color: theme.palette.themePrimary
-    },
-
-    rootPressed: {
-      color: theme.palette.black,
-    },
-
-    iconPressed: {
-      color: theme.palette.themeDarker
-    },
-
-    rootDisabled: {
-      color: theme.palette.neutralTertiary,
-      backgroundColor: 'transparent'
-    },
-
-    rootChecked: {
-      color: theme.palette.black,
-    },
-
-    iconChecked: {
-      color: theme.palette.themeDarker
-    },
-
-    flexContainer: {
-      justifyContent: 'flex-start'
-    },
-
-    icon: {
-      color: theme.palette.themeDarkAlt
-    },
-
-    iconDisabled: {
-      color: 'inherit'
-    },
-
+import { concatStyleSets, FontWeights } from '../../../Styling';
+import { IButtonBaseStyleProps, IButtonBaseStyles } from '../_base/Button.base.types';
+
+import { getButtonBaseStyles } from '../_base/Button.base.styles';
+
+
+export const getStyles = (props: IButtonBaseStyleProps): IButtonBaseStyles => {
+
+  const { theme, expanded, disabled, checked } = props;
+
+  const DEFAULT_BUTTON_HEIGHT = '40px';
+  const DEFAULT_PADDING = '0 4px';
+
+  let buttonStyles: IButtonBaseStyles = {
+    button: [
+      {
+        justifyContent: 'flex-start',
+        padding: DEFAULT_PADDING,
+        height: DEFAULT_BUTTON_HEIGHT,
+        color: theme.palette.neutralPrimary,
+        backgroundColor: 'transparent',
+        selectors: {
+          ':hover': {
+            color: theme.palette.themePrimary,
+          },
+          ':active': {
+            color: theme.palette.black,
+          }
+        }
+      },
+      checked && {
+        color: theme.palette.black,
+      },
+      disabled && {
+        color: theme.palette.neutralTertiary,
+        backgroundColor: 'transparent'
+      }
+    ],
+
+    icon: [
+      {
+        color: theme.palette.themeDarkAlt,
+        selectors: {
+          ':hover': {
+            color: theme.palette.themePrimary
+          },
+          ':active': {
+            color: theme.palette.themeDarker
+          }
+        }
+      },
+      checked && {
+        color: theme.palette.themeDarker
+      },
+      disabled && {
+        color: 'inherit'
+      }
+    ],
     menuIcon: {
       color: theme.palette.neutralSecondary
     },
 
     textContainer: {
       flexGrow: 0
     }
-
   };
 
-  return concatStyleSets(baseButtonStyles, actionButtonStyles, customStyles)!;
-});
\ No newline at end of file
+  const baseStyles: IButtonBaseStyles = getButtonBaseStyles(props);
+
+  return concatStyleSets(
```

Comment:
Do you think adding Jest snapshots for the styles of each variant would be a good idea? I imagine VR tests already exist for all these components, but having snapshots for each variant would let people know what components they have to manually validate when getButonBaseStyles is updated.